python/tank/bootstrap/async.py

```diff
@@ -0,0 +1,278 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+# Import Qt without having to worry about the version to use.
+from ..util.qt_importer import QtImporter
```

Comment:
That would imply that the auth framework would depend on the platform module, which would introduce another circular dependencym which I don't think is a wise-idea. After all, we already have a circular dependency between util and auth.

I'd like to have a discussion to see if we could instead move qt outside of platform and make it more official actually. With the QtImporter, maybe we can now act as if Qt supported by core and that an engine only wraps certain parts of Qt in certain DCCs instead of defining it. We then introduce sgtk.util.qt or sgtk.qt and sgtk.platform.qt in there for backwards compatibility.